pkg/client/group_mgmt_client.go

```diff
@@ -384,3 +405,62 @@ func unwrapError(body []byte) (string, error) {
 	}
 	return errResp, nil
 }
+
+// processAsyncResponse: process http code 202 response
+func processAsyncResponse(client *GroupMgmtClient, body []byte) (interface{}, error) {
```

Comment:
It looks like we are returning a job ID.  Should the return type be string?

Reply:
It’s better to return nil and handle nil check rather empty string check .

Reply:
Return a *string, then?

Reply:
There are multiple ways to do the same thing. is there any issue with interface here ?

Reply:
I think interface{} is for when you don't know what type will be returned.  If you know and you aren't telling me, well, that just seems kind of mean.  :)

Really though, I feel the best approach is to make the return type a string and to change the logic at line 128/158 to check for err == nil instead of id != nil.

Reply:
There is no hard rule that you need use interface only when you don't know the concrete type. you can use interface to write clean code . Consider this that you are returning string . You will have to write if st != "" {} such code . Here I am using interface to pass either string or nil without using a *string or string . In the caller i just need to check if it is nil or not . Clean code.

Reply:
Ok i accept this change.

Reply:
Interesting, I got an email when you added a comment saying you accept the change, but not when you edited the comment as above.

In the call stack you reference, GetJobObjectSet().GetObject() actually returns a *Job, not an interface{}.  The ObjectId is then extracted as a *string.

Both processAsyncResponse() and waitForJobResults() return an interface{} and an error, and I think both should be changed to return a string instead.  If the error is not nil, the string should be ignored, so there is no need to worry about the empty string as a special value.

The example code above is not compelling to me, because you don't explain what it means if the caller passes in a.non-string value.

Reply:
empty interfaces{} accepts any type of value. 
Moreover, this is a minor comment and shouldn't hold back feature merge.

Reply:
I agree it is a minor comment.  I was under the impression that another commit was coming, based on the comment I received in email.  I didn't know to look for an edited comment.